go/engine/pgp_purge.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2016 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package engine
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol"
+)
+
+// PGPPurge is an engine.
+type PGPPurge struct {
+	libkb.Contextified
+	arg       keybase1.PGPPurgeArg
+	me        *libkb.User
+	filenames []string
+}
+
+// NewPGPPurge creates a PGPPurge engine.
+func NewPGPPurge(g *libkb.GlobalContext, arg keybase1.PGPPurgeArg) *PGPPurge {
+	return &PGPPurge{
+		Contextified: libkb.NewContextified(g),
+		arg:          arg,
+	}
+}
+
+// Name is the unique engine name.
+func (e *PGPPurge) Name() string {
+	return "PGPPurge"
+}
+
+// GetPrereqs returns the engine prereqs.
+func (e *PGPPurge) Prereqs() Prereqs {
+	return Prereqs{
+		Device: true,
+	}
+}
+
+// RequiredUIs returns the required UIs.
+func (e *PGPPurge) RequiredUIs() []libkb.UIKind {
+	return []libkb.UIKind{}
+}
+
+// SubConsumers returns the other UI consumers for this engine.
+func (e *PGPPurge) SubConsumers() []libkb.UIConsumer {
+	return []libkb.UIConsumer{
+		&SaltpackEncrypt{},
+	}
+}
+
+// Run starts the engine.
+func (e *PGPPurge) Run(ctx *Context) error {
+	me, err := libkb.LoadMe(libkb.NewLoadUserPubOptionalArg(e.G()))
+	if err != nil {
+		return err
+	}
+	e.me = me
+
+	for _, bundle := range e.me.GetActivePGPKeys(false) {
```

Comment:
I can change it, but before I do:  

1. What is the utility of exporting inactive PGP keys?  They are already unable to use them with keybase.  Wouldn't it be slightly confusing to get these keys again?
2. There can be multiple blocks for the same key.  The skb keyring uses last writer wins when making the index.  If iterating over each block, we could export multiple versions.  Should it iterate over the index instead?